components/support/jwcrypto/src/encdec.rs

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{EncryptorDecryptorError, JwCryptoError, Jwk};
+use serde::{de::DeserializeOwned, Serialize};
+use std::marker::PhantomData;
+
+/// High-level struct for handling Encryption/Decryption
+///
+/// This struct wraps the jwcrypto functionality in a convenient package.  Also, it uses
+/// `EncryptorDecryptorError`, which includes a description that can help track down where
+/// encryption errors are happening.
+///
+/// EncryptorDecryptor takes a generic Error parameter that it uses for its results.  This can be
+/// anything that implement `From<EncryptorDecryptorError>`, typically it's the internal error type
+/// for a crate.
+pub struct EncryptorDecryptor<E = EncryptorDecryptorError> {
+    jwk: Jwk,
+    phantom: PhantomData<E>,
+}
+
+impl<E: From<EncryptorDecryptorError>> EncryptorDecryptor<E> {
+    /// Create a key that can be used to construct an EncryptorDecryptor
+    pub fn create_key() -> Result<String, E> {
+        let key = crate::Jwk::new_direct_key(None).to_encdec_result("create key")?;
```

Comment:
nit: is there a way for us to move around the magic strings? either make them an enum `EncryptionDecryptionReason`, or maybe consts? - thoughts?

Reply:
The issue is that `EncryptorDecryptorError` always expects a string.  I don't think it would be too hard to clean this up, but let's do it as a follow-up (#5493).